src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java

```diff
@@ -0,0 +1,218 @@
+    {
+        ArgumentState argumentState = argumentStateByActionArgumentNameMap.get(actionArgument.getName());
+        if (null == argumentState)
+        {
+            argumentState = new ArgumentState();
+            argumentStateByActionArgumentNameMap.put(actionArgument.getName(), argumentState);
+        }
+        argumentState.value = newValue;
+        argumentState.valueIsSet = true;
+    }
+
+    private List<ActionArgument> getAllArgumentsList()
+    {
+        return getAllArgumentsList(actionMenu.getActionArgumentList());
+    }
+
+    private List<ActionArgument> getAllArgumentsList(@NotNull final List<ActionArgument> currentActionArgumentList)
+    {
+        final List<ActionArgument> allActionArgumentsList = new ArrayList<>();
+        for (final ActionArgument actionArgument : currentActionArgumentList)
+        {
+            allActionArgumentsList.add(actionArgument);
+            if (isValueSet(actionArgument))
+            {
+                allActionArgumentsList.addAll(getAllArgumentsList(actionArgument.getActionArgumentList()));
+            }
+        }
+        return allActionArgumentsList;
+    }
+
+    public Colony getColonyForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.COLONY == actionArgument.getType())
+                {
+                    return (Colony) getValue(actionArgument);
+                }
+            }
+        }
+        return null;
+    }
+
+    public EntityPlayerMP getPlayerForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.PLAYER == actionArgument.getType())
+                {
+                    return (EntityPlayerMP) getValue(actionArgument);
+                }
+            }
+        }
+        return null;
+    }
+
+    public CitizenData getCitizenForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.CITIZEN == actionArgument.getType())
+                {
+                    return (CitizenData) getValue(actionArgument);
+                }
+            }
+        }
+        return null;
+    }
+
+    /*
+     * Suppressing Sonar Rule squid:S2447
+     * This rule complains about returning null for a Boolean method.
+     * But in this case the rule does not apply because
+     * We are returning null to indicate that no boolean argument value was set.
+     */
+    @SuppressWarnings({"squid:S2447"})
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2447' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)